### PR TITLE
fix: error handling in testsuite execution

### PR DESCRIPTION
**Pull request by exu**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: cmd/kubectl-testkube/commands/testsuites/run.go [hunk 1]
```diff
@@ -140,11 +140,17 @@ func NewRunTestSuiteCmd() *cobra.Command {
...
-						for execution := range executionCh {
-							ui.ExitOnError("watching test execution", err)
+						watchResp := client.WatchTestSuiteExecution(execution.Id)
+						ui.ExitOnError("watching test execution", err)
+
+						for resp := range watchResp {
+							if err != nil {
+								ui.Err(err)
+								return
+							}
```
Comment (rangoo94) [inline cmd/kubectl-testkube/commands/testsuites/run.go:150]: Where this `err` comes from?

**Review by vsukhin — APPROVED** [review]

File: cmd/kubectl-testkube/commands/testsuites/watch.go [hunk 2]
```diff
@@ -26,10 +26,10 @@ func NewWatchTestSuiteExecutionCmd() *cobra.Command {
 			startTime := time.Now()
 
 			executionID := args[0]
-			executionCh, err := client.WatchTestSuiteExecution(executionID)
-			for execution := range executionCh {
+			watchResp, err := client.WatchTestSuiteExecution(executionID)
+			for resp := range watchResp {
 				ui.ExitOnError("watching test execution", err)
```
Comment (rangoo94) [inline cmd/kubectl-testkube/commands/testsuites/watch.go:31]: `err` is not a pointer

Comment (rangoo94) [reply cmd/kubectl-testkube/commands/testsuites/watch.go:31]: It should use `resp.Error`

File: cmd/kubectl-testkube/commands/testsuites/watch.go [hunk 3]
```diff
@@ -26,10 +26,10 @@ func NewWatchTestSuiteExecutionCmd() *cobra.Command {
 			startTime := time.Now()
 
 			executionID := args[0]
-			executionCh, err := client.WatchTestSuiteExecution(executionID)
-			for execution := range executionCh {
+			watchResp, err := client.WatchTestSuiteExecution(executionID)
```
Comment (rangoo94) [inline cmd/kubectl-testkube/commands/testsuites/watch.go:29]: There is no `err` returned

File: cmd/kubectl-testkube/commands/testsuites/run.go [hunk 4]
```diff
@@ -140,11 +140,17 @@ func NewRunTestSuiteCmd() *cobra.Command {
 
 				if execution.Id != "" {
 					if watchEnabled && len(args) > 0 {
-						executionCh, err := client.WatchTestSuiteExecution(execution.Id)
-						for execution := range executionCh {
-							ui.ExitOnError("watching test execution", err)
+						watchResp := client.WatchTestSuiteExecution(execution.Id)
+						ui.ExitOnError("watching test execution", err)
```
Comment (rangoo94) [inline cmd/kubectl-testkube/commands/testsuites/run.go:144]: There is no `err` here

File: cmd/kubectl-testkube/commands/testsuites/run.go [hunk 5]
```diff
@@ -140,11 +140,17 @@ func NewRunTestSuiteCmd() *cobra.Command {
...
+
+						for resp := range watchResp {
+							if err != nil {
```
Comment (rangoo94) [inline cmd/kubectl-testkube/commands/testsuites/run.go:147]: We should use `resp.Error`

**Review by rangoo94 — APPROVED** [review]

**Pull request merged**